Convert VPopover story to MDX (#1089)
WordPress/openverse: frontend/src/components/VPopover/meta/VPopover.stories.mdx

import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Story,
} from "@storybook/addon-docs"

import { placements as popoverPlacements } from "@popperjs/core"
import { log } from "~/utils/console"

import VPopover from "~/components/VPopover/VPopover.vue"
import VButton from "~/components/VButton.vue"

<Meta
  title="Components/VPopover"
  components={VPopover}
  argTypes={{
    hideOnEsc: { control: { type: "boolean" } },
    hideOnClickOutside: { control: { type: "boolean" } },
    autoFocusOnShow: { control: { type: "boolean" } },
    autoFocusOnHide: { control: { type: "boolean" } },
    placement: { options: [...popoverPlacements], control: { type: "radio" } },
    label: { control: { type: "text" } },
    labelledBy: { control: { type: "text" } },
    close: { action: "close" },
    open: { action: "open" },
    popoverItems: { control: { type: "number" } },
  }}
/>

export const DefaultPopoverStory = (args) => ({
  template: `
    <div>
      <p>

>         This story is configured to log when the popover opens or closes. Inspect the console output to observe this behavior.

Reviewer: ```suggestion
        This story is configured to log when the popover opens or closes. Inspect the Actions tab to observe this behavior.
```

import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Story,
} from "@storybook/addon-docs"

import { placements as popoverPlacements } from "@popperjs/core"
import { log } from "~/utils/console"

import VPopover from "~/components/VPopover/VPopover.vue"
import VButton from "~/components/VButton.vue"

> <Meta title="Components/VPopover" components={VPopover} argTypes={{}} />
> 
> export const popoverArgTypes = {
>   hideOnEsc: { control: { type: "boolean" } },
>   hideOnClickOutside: { control: { type: "boolean" } },
>   autoFocusOnShow: { control: { type: "boolean" } },
>   autoFocusOnHide: { control: { type: "boolean" } },
>   placement: { options: [...popoverPlacements], control: { type: "radio" } },
>   label: { control: { type: "text" } },
>   labelledBy: { control: { type: "text" } },
>   close: { action: "close" },
>   open: { action: "open" },
> }

Reviewer 2: Instead of repeating arg types in each story, they can be set for the entire component.

```suggestion
<Meta title="Components/VPopover" components={VPopover} argTypes={{
  hideOnEsc: { control: { type: "boolean" } },
  hideOnClickOutside: { control: { type: "boolean" } },
  autoFocusOnShow: { control: { type: "boolean" } },
  autoFocusOnHide: { control: { type: "boolean" } },
  placement: { options: [...popoverPlacements], control: { type: "radio" } },
  label: { control: { type: "text" } },
  labelledBy: { control: { type: "text" } },
  close: { action: "close" },
  open: { action: "open" },
}} />
```

      </template>
    </VPopover>
  `,
  components: { VPopover, VButton },
  setup() {
    return { args }
  },
})

> export const TwoPopoverStory = (args) => ({

Reviewer 2: `TwoPopoverStory` and `SinglePopoverStory` can be merged using a prop and `v-for` to generate as many popovers as needed. But that's an improvement, not a blocker.

    </div>
  `,
  components: { VPopover, VButton },
  setup() {
    return { args }
  },
})

<Description of={VPopover} />

<ArgsTable of={VPopover} />

> ## Default VPopover

Reviewer 2: The 'VPopover' portion is implied in all `H2`s.

```suggestion
## Default
```

<Canvas>

>   <Story name="Default" argTypes={popoverArgTypes}>

Reviewer 2: If the arg types are repeated, they can be moved to the `<Meta>` tag at the top of the file.

```suggestion
  <Story name="Default" >
```
